### ISSUE TITLE
subgraph data is incorrectly created/overriden

### ISSUE THREAD

**Issue by yarrmateys** [TES5Edit/TES5Edit#864]
**What is the version of xEdit you are using?**
FO4Edit 4.1.3c EXTREMELY EXPERIMENTAL (768E0C49) starting session 2020-09-17 22:38:46

**Describe the bug**
attempting to override a race will damage the subgraph data, causing all the npcs of said race to t-pose ingame. 4.1.3 will show it to be identical, but 4.0.3 shows it to be all wrong. using 4.0.3 to fix the override produces functional npcs again.

**To reproduce**
1. override a race, say feralghoulrace (0006B4EC) using 4.1.3, and add the plugin to load order
2. go ingame, encounter/spawn an npc of said race
3. fix the override using 4.0.3
4. try again

**Expected behavior**
expected it to override the record like 4.0.3 did, so the animations would remain untouched.

**Comment by ElminsterAU** [comment]
@eckserah, could I please ask you to have a look the loader code for this too when you have some time?

I believe that it actually works completely different than what our record definitions currently express (and can express).

SubRecordStructs, as defined in xEdit, are triggered by the presence of the first contained subrecord, and then consume further subrecords until there is one that doesn't fit in anymore.

So with this definition (which is indentical between 4.0.x and 4.1.x code): 
```
    wbRArray('Subgraph Data',
      wbRStruct('Data', [
        wbString(SGNM, 'Behaviour Graph'),
        wbRArray('Actor Keywords', wbFormIDCk(SAKD, 'Keyword', [KYWD])),
        wbRArray('Target Keywords', wbFormIDCk(STKD, 'Keyword', [KYWD])),
        wbRArray('Animation Paths', wbString(SAPT, 'Path'), cpNormal, True),
        // Values greater than $10000 sets a bool. Reading this "closes" the current record.
        wbStruct(SRAF, 'Flags', [
          wbInteger('Role', itU16, wbEnum([
            {0} 'MT',
            {1} 'Weapon',
            {2} 'Furniture',
            {3} 'Linking Only',
            {4} 'Pipboy'
          ])),
          wbInteger('Perspective', itU16, wbEnum([
            '3rd',
            '1st'
          ]))
        ], cpNormal, True)
      ], [], cpNormal, False, nil, True)
    ),
```
Every SGNM is seen as the start of a new Data entry, and everything until the next SGNM goes into that entry.

What I believe the loader code actually does is collect information from all of these subrecords as it finds them, and then finishes the data entry when it hits an SRAF. Which is something that our current record definitions can't express.

Looking at the subrecords as they occur in Fallout4.esm for that record here, what the parser in 4.1.x, correctly according to the record definitions, reads is:
```
SGNM SAPT SRAF SAKD 
SGNM SAPT SAPT SRAF SAKD 
SGNM SAPT SAPT SRAF SAKD SAKD 
SGNM SAPT SAPT SRAF SAKD 
SGNM SAPT SRAF SAKD 
SGNM SAPT SRAF SAKD SAKD 
SGNM SAPT SRAF SAKD SAKD 
SGNM...
```
and what I think the data is actually saying is:
```
SGNM SAPT SRAF 
SAKD SGNM SAPT SAPT SRAF 
SAKD SGNM SAPT SAPT SRAF
SAKD SAKD SGNM SAPT SAPT SRAF 
SAKD SGNM SAPT SRAF 
SAKD SGNM SAPT SRAF 
SAKD SAKD SGNM SAPT SRAF 
...
```
Because of a bug in the 4.0.x codebase (which, *incorrectly*, terminated reading an "allow unordered" subrecord struct the moment it hit the last defined subrecord for that struct), the 4.0.x codebase "accidentally" parses this in a way which makes it behave like the 2nd listing above. But it's actually incorrect based on the record definitions (and leads to wrong parsing in other places). 

If you could confirm that the engine treats SRAF as "closer" instead of SGNM as "opener", I can add explicit support for that in the 4.1.x code base and the record definition can be appropriately flagged.

**Comment by ElminsterAU** [comment]
should be fixed with the commit above (will be in 4.1.3h)

I've operated on the assumption that what I said above is true and introduced a dfTerminator DefFlag which can be added to members of a SubRecordStruct. By adding this flag, parsing of the SubRecordStruct will stop after adding any member with that flag.

This now results in an override that when looked at with 4.0.x is identical.
